src/main/java/com/marianbastiurea/lifeofbees/Hive.java

```diff
@@ -187,6 +207,41 @@ public void checkAndAddEggsToBees() {
                 numberOfBees += eggsBatch.getNumberOfEggs();
             }
         }
+
+    }
+
+    public void addEggsFrame(Hive hive, List<EggsBatch> eggsBatches) {
+        int maxEggPerFrame = 6400;
+        int totalNumberOfEggsPerHiveAtBeginning = hive.getNumberOfEggs();
+        System.out.println("Total number of eggs at beginning: " + totalNumberOfEggsPerHiveAtBeginning);
+        System.out.println("number of eggsFrame " + hive.getNumberOfEggsFrame());
+        System.out.println(" maximum number of eggs for initial eggs frame " + maxEggPerFrame * hive.getNumberOfEggsFrame());
+        int totalNumberOfEggsPerHive = totalNumberOfEggsPerHiveAtBeginning;
+        do {
+          //  for (EggsBatch eggsBatch : eggsBatches) {
+
+            System.out.println("egge batched is"+hive.getEggsBatches());
+               // totalNumberOfEggsPerHive +=hive.getEggsBatches()
+                System.out.println("Total number of eggs with eggsBatch: " + totalNumberOfEggsPerHive);
+              //  System.out.println("date when initial eggs frame are full"+eggsBatches.getNumberOfEggs());
+           // }
+        //    System.out.println("Total number of eggs with eggsBatch: " + totalNumberOfEggsPerHive);
+           // System.out.println("date when initial eggs frame are full"+eggsBatch.getCreationDate());
+        } while (totalNumberOfEggsPerHive < maxEggPerFrame * hive.getNumberOfEggsFrame());
+        if (hive.getNumberOfEggsFrame() < 6) {
+            int numberOfFramesToAdd = 6 - hive.getNumberOfEggsFrame();
+            System.out.println("you can insert another eggsFrame" + numberOfFramesToAdd);
+            Queen queen = new Queen(hive.getAgeOfQueen());
```

Comment:
Dont create a new queen, get the existing queen from the hive.

Reply:
done